packages/twenty-front/src/modules/object-record/record-show/hooks/useRecordShowPagePagination.ts

```diff
@@ -0,0 +1,253 @@
+/* eslint-disable @nx/workspace-no-navigate-prefer-link */
+import { useMemo, useState } from 'react';
+import {
+  useLocation,
+  useNavigate,
+  useParams,
+  useSearchParams,
+} from 'react-router-dom';
+import { useSetRecoilState } from 'recoil';
+
+import { useObjectMetadataItem } from '@/object-metadata/hooks/useObjectMetadataItem';
+import { formatFieldMetadataItemsAsFilterDefinitions } from '@/object-metadata/utils/formatFieldMetadataItemsAsFilterDefinitions';
+import { formatFieldMetadataItemsAsSortDefinitions } from '@/object-metadata/utils/formatFieldMetadataItemsAsSortDefinitions';
+import { generateDepthOneRecordGqlFields } from '@/object-record/graphql/utils/generateDepthOneRecordGqlFields';
+import { useFindManyRecords } from '@/object-record/hooks/useFindManyRecords';
+import { turnSortsIntoOrderBy } from '@/object-record/object-sort-dropdown/utils/turnSortsIntoOrderBy';
+import { lastShowPageRecordIdState } from '@/object-record/record-field/states/lastShowPageRecordId';
+import { turnObjectDropdownFilterIntoQueryFilter } from '@/object-record/record-filter/utils/turnObjectDropdownFilterIntoQueryFilter';
+import { useRecordIdsFromFindManyCacheRootQuery } from '@/object-record/record-show/hooks/useRecordIdsFromFindManyCacheRootQuery';
+import { usePrefetchedData } from '@/prefetch/hooks/usePrefetchedData';
+import { PrefetchKey } from '@/prefetch/types/PrefetchKey';
+import { View } from '@/views/types/View';
+import { mapViewFiltersToFilters } from '@/views/utils/mapViewFiltersToFilters';
+import { mapViewSortsToSorts } from '@/views/utils/mapViewSortsToSorts';
+import { isNonEmptyString } from '@sniptt/guards';
+import { capitalize } from '~/utils/string/capitalize';
+
+export const findView = ({
+  viewId,
+  objectMetadataItemId,
+  views,
+}: {
+  viewId: string | null;
+  objectMetadataItemId: string;
+  views: View[];
+}) => {
+  if (!viewId) {
+    return views.find(
+      (view: any) =>
+        view.key === 'INDEX' && view?.objectMetadataId === objectMetadataItemId,
+    ) as View;
+  } else {
+    return views.find(
+      (view: any) =>
+        view?.id === viewId && view?.objectMetadataId === objectMetadataItemId,
+    ) as View;
+  }
+};
+
+export const useRecordShowPagePagination = (
```

Comment:
I feel this could be in a separate module "navigate-in-view" or something like this and could be re-used in other places later